inst/unitTests/runit.Matrix.R

```diff
@@ -171,4 +171,16 @@ if (.runThisTest) {
       checkEquals( m[,1], m[,2] )
     }
 
+    test.IntegerMatrix.accessor.with.bounds.checking <- function() {
+        m <- matrix(seq(1L, 12, by=1L), nrow=4L, ncol=3L)
+        checkEquals(mat_access_with_bounds_checking(m, 0, 0), 1)
+        checkEquals(mat_access_with_bounds_checking(m, 1, 2), 10)
+        checkEquals(mat_access_with_bounds_checking(m, 3, 2), 12)
+        checkException(mat_access_with_bounds_checking(m, 4, 2) , msg = "index out of bounds not detected" )
```

Comment:
I guess you are correct -- if it passes we just get `TRUE`.   But if you look at our code the msg field, when we set it all :), mostly just states where this came from.  So  "matrix bounds check" would be fine, and s/matrix/vector/ for the other one.

Not that important.  I'll try merge this maybe late this evening if I get to it.  It now needs a manual intervention because ChangeLog changed in the meantime ...